rotkehlchen/chain/ethereum/modules/lido/decoder.py

```diff
@@ -0,0 +1,526 @@
+    ) -> DecodingOutput:
+        """Decode steth wrapping in wsteth from TransferShares log event"""
+
+        user_address = hex_or_bytes_to_address(context.tx_log.topics[1])
+
+        #  we need to retrieve the transferred stETH amount deterministically.
+        #  expected log index for the transferred steth is -1
+        #  expected log index for the transferred wsteth is -3
+        expected_steth_transfer_index = context.tx_log.log_index - 1
+        expected_wsteth_transfer_index = context.tx_log.log_index - 3
+        paired_wsteth_ratio = None
+        sth_wrapped_amount = None
+        wsteth_received_amount = None
+
+        for tx_log in context.all_logs:
+            if (
+                tx_log.log_index == expected_steth_transfer_index and
+                tx_log.topics[0] == ERC20_OR_ERC721_TRANSFER and
+                tx_log.address == steth_evm_address and
+                hex_or_bytes_to_address(tx_log.topics[1]) == user_address and  # from
+                hex_or_bytes_to_address(tx_log.topics[2]) == wsteth_evm_address  # to # wsteth
+            ):
+                sth_wrapped_amount = asset_normalized_value(
+                    amount=hex_or_bytes_to_int(tx_log.data[:32]),
+                    asset=A_STETH.resolve_to_crypto_asset(),
+                )
+            elif (
+                tx_log.log_index == expected_wsteth_transfer_index and
+                tx_log.topics[0] == ERC20_OR_ERC721_TRANSFER and
+                tx_log.address == wsteth_evm_address and
+                hex_or_bytes_to_address(tx_log.topics[1]) == ZERO_ADDRESS and  # from
+                hex_or_bytes_to_address(tx_log.topics[2]) == user_address  # to # wsteth
+            ):
+                wsteth_received_amount = asset_normalized_value(
+                    amount=hex_or_bytes_to_int(tx_log.data[:32]),
+                    asset=A_WSTETH.resolve_to_crypto_asset(),
+                )
+
+            if sth_wrapped_amount and wsteth_received_amount:
+                paired_wsteth_ratio = FVal(sth_wrapped_amount) / FVal(wsteth_received_amount)
+                break
+
+        if paired_wsteth_ratio is None:
+            return DEFAULT_DECODING_OUTPUT
+
+        in_event = None
+        out_event = None
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset == A_STETH and
+                event.balance.amount == sth_wrapped_amount and
+                event.address == wsteth_evm_address
+            ):
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.counterparty = CPT_LIDO_ETH
+                event.notes = f'Sent {sth_wrapped_amount} {A_STETH.symbol_or_name()} for wrapping at wstETH ratio {paired_wsteth_ratio}'  # noqa: E501
```

Comment:
Something to keep in mind for all events. I won't comment on all others.

1. Always present tense
2. no need to run function when needed. We know symbol of the token.
3. Why do you need the ratio? Just informational purposes?
```suggestion
                event.notes = f'Send {sth_wrapped_amount} stETH for wrapping at wstETH ratio {paired_wsteth_ratio}'  # noqa: E501
```

Reply:
1 and 2, I will fix.
For 3, imho it is important to track the valute of the ratio at the moment the user obtained it, because it is not a fixed ratio, but it grows with time and stETH tokens obtained by Lido staking. With that, user can know its current gaining, looking at the original and current ratio.